frontend/src/metabase/lib/redux/typed-utils.ts

```diff
@@ -11,13 +12,26 @@ interface ThunkConfig {
 export const createAsyncThunk =
   createAsyncThunkOriginal.withTypes<ThunkConfig>();
 
+type AwaitedIfPromise<T> = T extends Promise<infer R> ? R : T;
+
 // similar to createAction but accepts a (redux-thunk style) thunk and dispatches based on whether
 // the promise returned from the thunk resolves or rejects, similar to redux-promise
-export function createThunkAction<TArgs extends any[]>(
-  actionType: string,
+export function createThunkAction<
```

Comment:
I wouldn't say this typing is perfect, but I think it's good enough to avoid having to manually write types.